### ISSUE TITLE
Question on adj constructed after removing loss edges from train_pos

### ISSUE THREAD

**Issue by hsinghuan** [Juanhui28/HeaRT#5]
Hi! I am new to OGB link prediction datasets so this question might seem naive.

It seems like you constructed adj from train_pos (split_edge["train"]["edge"]) after removing the loss edge of the mini-batch as follows.

```python
######################### remove loss edges from the aggregation
mask = torch.ones(train_pos.size(0), dtype=torch.bool).to(train_pos.device)
mask[perm] = 0

train_edge_mask = train_pos[mask].transpose(1,0)
train_edge_mask = torch.cat((train_edge_mask, train_edge_mask[[1,0]]),dim=1)

if pos_train_weight != None:
    edge_weight_mask = pos_train_weight[mask]
    edge_weight_mask = torch.cat((edge_weight_mask, edge_weight_mask), dim=0).to(torch.float)
   

else:
    edge_weight_mask = torch.ones(train_edge_mask.size(1)).to(torch.float).to(train_pos.device)

adj = SparseTensor.from_edge_index(train_edge_mask, edge_weight_mask, [num_nodes, num_nodes]).to(train_pos.device)
    
###################
```

However, wouldn't the training message passing graph miss some edges in the original graph object (data.adj_t) as not all edges in data.adj_t are in split_edge["train"]["edge"]? Is that intentional?

I'm not sure if I miss something but I'm curious about this. Thanks!

**Comment by Juanhui28** [comment]
Hi, thanks for the interest!  This piece of code is a simple trial to mitigate the data shift problem in link prediction ( as mentioned in this paper: https://arxiv.org/pdf/2211.15899).  But according to my empirical experience, removing the edges or not doesn't impact performance so much and would be time consuming on large-scale datasets. Feel free to just
remove this part.

Thanks!


hsinghuan ***@***.***> 于2024年4月24日周三 17:44写道：

**Comment by hsinghuan** [comment]
Got it. Thanks for the prompt reply!
